### PR TITLE
Added -Wall and -Werror flags

### LINKED ISSUE #127: Add Wall and -Werror

### PR DESCRIPTION
**Pull request by Kfeavel**

### CODE REVIEW

File: kernel/dev/rtc/rtc.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ uint8_t px_rtc_get_register(int reg) {
 void px_read_rtc() {
     // Previous values from RTC
     // Used as a cache to check if we should update
-    uint8_t century, last_second, last_minute, 
+    uint8_t px_rtc_century, last_second, last_minute,
```
Comment (micahswitzer) [inline kernel/dev/rtc/rtc.cpp:65]: Are these century changes related to the new flags?

File: kernel/lib/bitmap.cpp [hunk 2]
```diff
@@ -17,7 +17,9 @@ size_t bitmap_find_first_range_clear(bitmap_t *bitmap, size_t size, size_t count
 #ifdef TESTING
         printf("i = %d, idx = 0x%08x, ofst = 0x%08x, check_lo = 0x%08x, "
                "check_hi = 0x%08x, check = 0x%08x, masked = 0x%08x\n",
-                i, idx, ofst, check_lo, check_hi, check, masked); 
+                i, idx, ofst, check_lo, check_hi, check, masked);
+#else
+        (void)masked;
```
Comment (micahswitzer) [inline kernel/lib/bitmap.cpp]: Just include the declaration and assignment inside the `ifdef`.

File: kernel/sys/panic.cpp [hunk 3]
```diff
@@ -100,7 +100,6 @@ void panic(registers_t *regs, const char *file, uint32_t line, const char *func)
     int rw = regs->err_code & 0x2;           // Write operation?
     int us = regs->err_code & 0x4;           // Processor was in user-mode?
     int reserved = regs->err_code & 0x8;     // Overwritten CPU-reserved bits of page entry?
-    int id = regs->err_code & 0x10;          // Caused by an instruction fetch?
```
Comment (micahswitzer) [inline kernel/sys/panic.cpp]: I feel like this would actually be useful information to see on the panic screen.

File: kernel/dev/rtc/rtc.cpp (hunk 1 above)
Comment (micahswitzer) [inline kernel/dev/rtc/rtc.cpp:65]: Are these changes related to adding those flags?

Comment (Kfeavel) [reply kernel/dev/rtc/rtc.cpp:65]: Yes. It said it could be uninitialized, so I decided it's best to just let it get its value since we hard coded the check anyways. RTC still gives the right date, so ¯\_(ツ)_/¯

File: kernel/lib/bitmap.cpp (hunk 2 above)
Comment (Kfeavel) [reply kernel/lib/bitmap.cpp]: Done.

File: kernel/dev/rtc/rtc.cpp (hunk 1 above)
Comment (Kfeavel) [reply kernel/dev/rtc/rtc.cpp:65]: Tangentially. I had to update the century RTC code and this wasn't named according to the style guide, so I updated it.

File: kernel/sys/panic.cpp (hunk 3 above)
Comment (Kfeavel) [reply kernel/sys/panic.cpp]: Resolved.

**Review by micahswitzer — APPROVED** [review]
